src/switch_core_media.c

```diff
@@ -5314,7 +5314,7 @@ static void check_stream_changes(switch_core_session_t *session, const char *r_s
 				switch_core_media_merge_sdp_codec_string(session, r_sdp, sdp_type, filter_codec_string);
 			}
 
-			if (switch_channel_test_flag(session->channel, CF_SECURE)) {
+			if (switch_channel_test_flag(session->channel, CF_SECURE) && switch_channel_test_flag(other_session->channel, CF_SECURE)) {
```

Comment:
I think we also need to check the other_session for CF_SECURE. For example if somehow the current session isn't secure but the partner session is secure then we still need to execute `switch_core_session_check_outgoing_crypto`. If both is secured then we will set the crypto_mode and call `switch_core_session_check_outgoing_crypto`